iroh-net/src/test_utils.rs

```diff
@@ -54,3 +60,68 @@ impl std::io::Write for TestWriter {
         std::io::stdout().flush()
     }
 }
+
+/// Runs a  DERP server with STUN enabled suitable for tests.
+///
+/// The returned `oneshot::Sender<()>` is a drop guard: the server will be shut down when
+/// dropped.  You could also explicitly send a value to achieve the same if you so desire.
+///
+/// The returned `u16` is the region ID of the DERP server in the returned [`DerpMap`], it
+/// is always `Some` as that is how the [`MagicEndpoint::connect`] API expects it.
+///
+/// [`MagicEndpoint::connect`]: crate::magic_endpoint::MagicEndpoint
+pub async fn run_derp_and_stun(
+    stun_ip: IpAddr,
+) -> Result<(DerpMap, Option<u16>, oneshot::Sender<()>)> {
```

Comment:
maybe make the oneshot an actual dropguard?

Reply:
it crossed my mind.  though i don't know what it would really gain other than a newtype.

maybe a nice version might be a type that's a dropguard as well as has a `.cleanup().await` that waits for tidy completion if you really want.  though i'm again wondering what the benefit is, each test gets a separate runtime anyway so as long as this cleanup is executed it's fine, even if it runs after the test.

so i really think it would be just a newtype, and i didn't really consider that worth it for a test.  but i guess we can do this.